src/caselawclient/models/documents/__init__.py

```diff
@@ -161,7 +161,15 @@ def public_uri(self) -> str:
         """
         :return: The absolute, public URI at which a copy of this document can be found
         """
-        return f"https://caselaw.nationalarchives.gov.uk/{self.uri}"
+        return f"https://caselaw.nationalarchives.gov.uk/{self.slug}"
+
+    @cached_property
+    def slug(self) -> str:
+        preferred_identifier = self.identifiers.preferred()
+        if preferred_identifier:
+            return preferred_identifier.url_slug
+        msg = f"No preferred identifier exists for {self.uri}"
+        raise RuntimeError(msg)
```

Comment:
This makes sense to me, we should never be trying to use this in a context where documents don't at least have an FCLID.